src/Domain/UseCase/CreateValetToken/CreateValetToken.ts

```diff
@@ -0,0 +1,56 @@
+import { inject, injectable } from 'inversify'
+import { isOperationPermitted } from '../../Operation/isOperationPermitted'
+import { ValetToken } from '../../ValetToken/ValetToken'
+import { CreateValetKeyDto } from './CreateValetTokenDto'
+import { CreateValetKeyResponse } from './CreateValetTokenResponse'
+import { UseCaseInterface } from '../UseCaseInterface'
+import { generateValetToken } from '../../ValetToken/generateValetToken'
+import TYPES from '../../../Bootstrap/Types'
+import { JwtSecret, ValetTokenSecret } from '../../ValetToken/ValetTokenGenerator'
+import { CrypterInterface } from '../../Encryption/CrypterInterface'
+import { createValetPayload } from '../../ValetToken/createValetPayload'
+
+@injectable()
+export class CreateValetToken implements UseCaseInterface {
+  constructor(
+    @inject(TYPES.JWT_SECRET) private jwtSecret: JwtSecret,
+    @inject(TYPES.VALET_TOKEN_SECRET) private valetTokenSecret: ValetTokenSecret,
+    @inject(TYPES.Crypter) private crypter: CrypterInterface,
+  ) {}
+
+  /**
+   * Given a request from the API Gateway containing auth data for a user and the requested file operations generates a valet key which can be used to perform these operations directly via the Files Service.
+   */
+  async execute(dto: CreateValetKeyDto): Promise<CreateValetKeyResponse> {
+    const { user, operation, resources, validityPeriod } = dto
+    const { permissions, uuid } = user
+
+    if (!isOperationPermitted({
```

Comment:
turn into `this.operationValidator.validate`

Reply:
Discussion here: https://github.com/standardnotes/files/pull/3#discussion_r598958158